crates/ra_hir_ty/src/infer/expr.rs

```diff
@@ -191,6 +233,9 @@ impl<'a> InferenceContext<'a> {
                     // FIXME handle break with value
                     self.infer_expr(*expr, &Expectation::none());
                 }
+                if let Some(ctxt) = self.breakables.last_mut() {
```

Comment:
I guess we might emit a diagnostic here? Seems like we can't do this syntactically, as it needs to happen post expansion

Reply:
I added the diagnostic. It's quite a bit of boilerplate :sweat:

Reply:
Yeah... 

Designing diagnostics infra seems like a task we can tackle now, as we actually have quite a few things we can report w/o false positives now...

Though, I must say I like how concise the emitting part looks